### PR TITLE
chore(windows platform): Windows support for ENVIRONMENT=true

### PR DESCRIPTION
**Pull request by Hoverbear**
This improves our Windows support more so `make build ENVIRONMENT=true VERBOSE=true` works golden on it.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -112,7 +113,7 @@ define ENVIRONMENT_EXEC
 			--interactive \
 			--env INSIDE_ENVIRONMENT=true \
 			--network host \
-			--mount type=bind,source=${PWD},target=/git/timberio/vector \
+			--mount type=bind,source=${CURRENT_DIR},target=/git/timberio/vector \
```
Comment (Hoverbear) [inline Makefile:116]: Weirdly, the Windows `make` does not support `${PWD}`.

File: scripts/environment/prepare.sh [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #! /usr/bin/env bash
 set -e -o verbose
 
+rustup toolchain install $(cat rust-toolchain)
```
Comment (Hoverbear) [inline scripts/environment/prepare.sh:4]: I'm not sure why, but on the Windows edge Docker this line is required.

**Review by jamtur01 — APPROVED** [review]
LGTM

**Comment by Hoverbear** [comment]
Hit by #3843

**Pull request merged**